src/helpers/fetchSearch.js

```diff
@@ -1,24 +1,43 @@
 import { api } from 'api'
 import getAccessionCodesQueryParam from './getAccessionCodesQueryParam'
+import getUniqElementsBy from './getUniqElementsBy'
 
-export default async (queryString) => {
+export default async (queryString, currentPage) => {
   const response = await api.search.get(queryString)
-  let accessionCodesResponse
+  let { results } = response
+  const { count: totalResults, facets } = response
   const accessionCodes = getAccessionCodesQueryParam(queryString.search)
 
-  if (accessionCodes) {
-    accessionCodesResponse =
-      (await Promise.all(
-        accessionCodes.map((code) =>
-          api.search.get({
-            search: [
-              `accession_code:${code}`,
-              `alternate_accession_code:${code}`
-            ]
-          })
-        )
-      )) || []
+  // makes requests for accession codes only from the first page
+  if (accessionCodes.length > 0 && currentPage === 1) {
+    const accessionCodesResponse = await Promise.all(
+      accessionCodes.map((code) =>
+        api.search.get({
+          search: [`accession_code:${code}`, `alternate_accession_code:${code}`]
+        })
+      )
+    )
+
+    const matchedAccessionCodes = []
+      .concat(...accessionCodesResponse.map((data) => data.results))
+      .map((result) => ({
+        ...result,
+        // adds the flag below to accession codes that matches the user's searched accession codes
+        // these are usually from accession code search
+        isMatchedAccessionCode: true
+      }))
+
+    if (matchedAccessionCodes.length > 0) {
+      // removes duplicates due to multiple requests for fetching accession codes
+      results = getUniqElementsBy(
+        [...matchedAccessionCodes, ...results],
+        (x) => x.accession_code
+      )
+    } else if (accessionCodes.length === 1) {
+      // returns no result if no match found for a single accession code that users searched for
+      results = []
+    }
```

Comment:
I'm going to open a ticket because this approach is ill-advised for the following reasons.

Given that:
- N is the number of accessions parsed from the query.
- X is the number of results per search result page.

1. This prepends the matched accession codes to the page of the results. That page states to the user that we are showing X results per page. With this implementation we will be showing X + N results on the page where N is the number displayed on the page.
1. This approach is unnecessarily expensive as it will create N + 1 queries per result when this could be 1.

I think this could be resolved with 2 changes.
1. Construct a query that uses the built in `accession_code__in` and `alternate_accession_code__in` query parameters. To fetch all experiments specifically mentioned in the query string.
1. Add some visual distinction between the augmented matches at the top and the actual search results. This would be a Deepa question.